daphne_server/src/roles/aggregator.rs

```diff
@@ -93,17 +247,32 @@ impl<S: Sync> DapAggregator<S> for crate::App {
 
     async fn taskprov_put(
         &self,
-        req: &DapRequest<S>,
+        req: &DapRequest<DaphneAuth>,
         task_config: DapTaskConfig,
     ) -> Result<(), DapError> {
-        todo!()
+        let task_id = req.task_id().map_err(DapError::Abort)?;
+
+        if self.service_config.role.is_leader() || req.taskprov.is_none() {
+            self.kv()
+                .put::<kv::prefix::TaskConfig>(task_id, task_config)
+                .await
+                .map_err(|e| fatal_error!(err = ?e))?;
+        } else {
+            self.kv()
+                .only_cache_put::<kv::prefix::TaskConfig>(task_id, task_config)
+                .await;
+        }
+        Ok(())
     }
 
     async fn get_task_config_for<'req>(
         &'req self,
         task_id: &'req TaskId,
     ) -> Result<Option<Self::WrappedDapTaskConfig<'req>>, DapError> {
-        todo!()
+        self.kv()
+            .get::<kv::prefix::TaskConfig>(task_id)
```

Comment:
Does this check cache for (and hit KV in case of miss)?

Reply:
All `get`s go through the cache